osu.Game/Screens/Select/SongSelect.cs

```diff
@@ -809,6 +809,98 @@ public override bool OnBackButton()
             return false;
         }
 
+        private void increaseSpeed()
+        {
+            var rateAdjustStates = ModSelect.AllAvailableMods.Where(pair => pair.Mod is ModRateAdjust);
+            var stateDoubleTime = ModSelect.AllAvailableMods.First(pair => pair.Mod is ModDoubleTime);
+            bool rateModActive = ModSelect.AllAvailableMods.Count(pair => pair.Mod is ModRateAdjust && pair.Active.Value) > 0;
+            const double stepsize = 0.05d;
+            double newRate = 1d + stepsize;
+
+            // If no mod rateAdjust mod is currently active activate DoubleTime with speed newRate
+            if (!rateModActive)
+            {
+                stateDoubleTime.Active.Value = true;
+                ((ModDoubleTime)stateDoubleTime.Mod).SpeedChange.Value = newRate;
+                return;
+            }
+
+            // Find current active rateAdjust mod and modify speed, enable DoubleTime if necessary
+            foreach (var state in rateAdjustStates)
```

Comment:
This should be asserting that there is only one rate adjust mod present rather than doing some weird loop.

Reply:
Done in [3fdbd73](https://github.com/ppy/osu/pull/28071/commits/3fdbd735ce063653bb92b7570df716e700a9b529)